spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/reader/ValueRowReader.scala

```diff
@@ -14,7 +14,10 @@ class ValueRowReader[T: TypeConverter](columnRef: ColumnRef) extends RowReader[T
     * @param row row fetched from Cassandra
     * @param rowMetaData: column names available in the `row` */
   override def read(row: Row, rowMetaData: CassandraRowMetadata): T =
-    converter.convert(GettableData.get(row, columnRef.cqlValueName))
+    converter.convert(GettableData.get(
+      row,
+      columnRef.cqlValueName,
+      rowMetaData.codecs(columnRef.cqlValueName)))
```

Comment:
I feel like having an interface for all the rowreader to have an initial "Setup" may be the key here so we don't have to do something like

```
  private var codec: Option[TypeCodec[AnyRef]] = None
  private var columnIndex: Option[Int] = None

  /** Reads column values from low-level `Row` and turns them into higher level representation.
    * @param row row fetched from Cassandra
    * @param rowMetaData: column names available in the `row` */
  override def read(row: Row, rowMetaData: CassandraRowMetadata): T = {
    if (codec.isEmpty) {
      codec = Some(rowMetaData.codecs(rowMetaData.namesToIndex(columnRef.cqlValueName)))
      columnIndex = Some(rowMetaData.namesToIndex(columnRef.cqlValueName))
    }
    converter.convert(GettableData.get(
      row,
      columnIndex.get,
      codec.get))
  }```